api_fhir_r4/converters/activityDefinitionConverter.py

```diff
@@ -119,301 +156,247 @@ def build_fhir_title(cls, fhir_activity_definition, imis_activity_definition):
+                                            cls.build_fhir_workflow(imis_activity_definition))
 
     @classmethod
-    def build_fhir_age(cls, imis_activity_definition):
-        adult = cls.build_fhir_adult(imis_activity_definition)
-        kid = cls.build_fhir_kid(imis_activity_definition)
+    def build_fhir_workflow(cls, imis_activity_definition):
         codeable_concept = CodeableConcept.construct()
+        codeable_concept.coding = []
 
-        if adult:
-            coding_adult = Coding.construct()
-            coding_adult.code = adult
-            coding_adult.display = "Adult"
-            if type(codeable_concept.coding) is not list:
-                codeable_concept.coding = [coding_adult]
-            else:
-                codeable_concept.coding.append(coding_adult)
-            codeable_concept.text = "Adult"
-
-        if kid:
-            coding_kid = Coding.construct()
-            coding_kid.code = kid
-            coding_kid.display = "Kid"
-            if type(codeable_concept.coding) is not list:
-                codeable_concept.coding = [coding_kid]
-            else:
-                codeable_concept.coding.append(coding_kid)
-            codeable_concept.text = "Adult or Kid"
+        codeable_concept.coding.append(cls.build_fhir_workflow_coding(imis_activity_definition.category))
+        codeable_concept.text = codeable_concept.coding[0].display
 
         return codeable_concept
 
     @classmethod
     def build_fhir_venue(cls, imis_activity_definition):
-        display = ""
-        if imis_activity_definition.care_type == "O":
-            display = "Out-patient"
-        if imis_activity_definition.care_type == "I":
-            display = "In-patient"
-        if imis_activity_definition.care_type == "B":
-            display = "Both"
-
         codeable_concept = CodeableConcept.construct()
-        coding_venue = Coding.construct()
-        coding_venue.code = imis_activity_definition.care_type
-        coding_venue.display = display
-        if type(codeable_concept.coding) is not list:
-            codeable_concept.coding = [coding_venue]
-        else:
-            codeable_concept.coding.append(coding_venue)
-        codeable_concept.text = "Clinical Venue"
+        codeable_concept.coding = []
+
+        ambulatory_venue = imis_activity_definition.care_type in [Service.CARE_TYPE_OUT_PATIENT, Service.CARE_TYPE_BOTH]
+        if ambulatory_venue:
+            ambulatory_coding = cls.build_fhir_venue_coding(Service.CARE_TYPE_OUT_PATIENT)
+            codeable_concept.coding.append(ambulatory_coding)
+
+        imp_venue = imis_activity_definition.care_type in [Service.CARE_TYPE_IN_PATIENT, Service.CARE_TYPE_BOTH]
+        if imp_venue:
+            imp_coding = cls.build_fhir_venue_coding(Service.CARE_TYPE_IN_PATIENT)
+            codeable_concept.coding.append(imp_coding)
+
+        codeable_concept.text = " or ".join([coding.display for coding in codeable_concept.coding])
```

Comment:
why not removing the coding if the coding.display is empty ? a complementary solution would be to use CODE if display is empty

Reply:
We need to define the code system used. useContext, parent item, cannot be code.